src/env.rs

```diff
@@ -106,10 +116,29 @@ impl Source for Environment {
                 key = key.replace(separator, ".");
             }
 
-            m.insert(
-                key.to_lowercase(),
-                Value::new(Some(&uri), ValueKind::String(value)),
-            );
+            let value = if self.try_parsing {
+                let string_value = Value::new(Some(&uri), ValueKind::String(value.clone()));
```

Comment:
There's a lot of `clone()`ing involved here. Can we somehow reduce it?

This line clones, and later we clone the `string_value` variable again in each `if-let` ... so in the worst-case, there are 5 `clone()` calls. I bet this can be improved!